ui/src/components/pages/Namespace/index.tsx

```diff
@@ -0,0 +1,228 @@
+import { useEffect, useState } from "react";
+import { NamespaceRowContent } from "./partials/NamespaceRowContent";
+import {
+  TableBody,
+  Table,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  TextField,
+  Autocomplete,
+  InputAdornment,
+  Popper,
+} from "@mui/material";
+import Box from "@mui/material/Box";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import SearchIcon from "@mui/icons-material/Search";
+import { notifyError } from "../../../utils/error";
+import { useSystemInfoFetch } from "../../../utils/fetchWrappers/systemInfoFetch";
+import { useNamespaceListFetch } from "../../../utils/fetchWrappers/namespaceListFetch";
+import { SetNamespaceList, SetStore } from "../../../localStore/SetStore";
+import { GetStore } from "../../../localStore/GetStore";
+
+import "./style.css";
+
+export function Namespaces() {
+  const [nsCookie, setNsCookie] = useState<string[]>([]);
+  const [nsCluster, setNsCluster] = useState<string[]>([]);
+  const [nsCombined, setNsCombined] = useState<any[]>([]);
+  const [namespace, setNamespace] = useState("");
+  const [value, setValue] = useState("");
+  const [disableSearch, setDisableSearch] = useState(false);
+  const { systemInfo, error: systemInfoError } = useSystemInfoFetch();
+  const { namespaceList: clusterNamespaces, error: namespaceListError } =
+    useNamespaceListFetch();
+
+  // check system info error
+  useEffect(() => {
+    if (systemInfoError) {
+      notifyError([
+        {
+          error: "Failed to fetch the system info",
+          options: { toastId: "ns-scope", autoClose: false },
+        },
+      ]);
+    }
+  }, [systemInfoError]);
+
+  // check cluster namespace fetching error
+  useEffect(() => {
+    if (namespaceListError && systemInfo && systemInfo?.namespaced === false) {
+      notifyError([
+        {
+          error: "Failed to fetch the available namespaces",
+          options: { toastId: "ns-cluster", autoClose: false },
+        },
+      ]);
+    }
+  }, [namespaceListError, systemInfo]);
+
+  // provides namespace search based on scoped installation
+  useEffect(() => {
+    if (systemInfo && systemInfo?.namespaced) {
+      setNamespace(systemInfo?.managedNamespace);
+      setDisableSearch(true);
+    } else if (systemInfo && !systemInfo?.namespaced) {
+      setDisableSearch(false);
+
+      // set namespace value for search box
+      const curr_ns = GetStore("curr_namespace");
```

Comment:
We can add curr_namespace as a global variable. In that way, when its used multiple times, its easy to manage